crates/timers/src/lib.rs

```diff
@@ -286,3 +287,201 @@ impl Future for TimeoutFuture {
         }
     }
 }
+
+/// A scheduled interval.
+///
+/// See `Interval::new` for scheduling new intervals.
+///
+/// Once scheduled, you can either `cancel` so that it ceases to fire or `forget`
+/// it so that it is un-cancel-able.
+#[must_use = "intervals cancel on drop; either call `forget` or `drop` explicitly"]
+pub struct Interval {
+    id: Option<i32>,
+    closure: Option<Closure<FnMut()>>,
+}
+
+impl Drop for Interval {
+    fn drop(&mut self) {
+        if let Some(id) = self.id {
+            window().clear_interval_with_handle(id);
+        }
+    }
+}
+
+impl fmt::Debug for Interval {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        f.debug_struct("Interval").field("id", &self.id).finish()
+    }
+}
+
+impl Interval {
+    /// Schedule an interval to invoke `callback` every `millis` milliseconds.
+    ///
+    /// # Example
+    ///
+    /// ```no_run
+    /// use gloo_timers::Interval;
+    ///
+    /// let timeout = Interval::new(1_000, move || {
+    ///     // Do something...
+    /// });
+    /// ```
+    pub fn new<F>(millis: u32, callback: F) -> Interval
+    where
+        F: 'static + FnOnce(),
```

Comment:
`FnOnce` is for closures that potentially consume values they close over and drop them. Therefore it is only safe to call them once, since callign them a second time will result in using already-dropped values. This makes sense for timers, which are invoked at most once.

Intervals on the other hand, are explicitly invoked many times at some frequency. Instead, we should use `FnMut`, which is a closure that is allowed to mutate its captured values, but not drop them.

https://doc.rust-lang.org/nightly/std/ops/trait.FnOnce.html
https://doc.rust-lang.org/nightly/std/ops/trait.FnMut.html
https://doc.rust-lang.org/nightly/book/ch13-01-closures.html

Reply:
Done!